js/src/components/shipping-rate-section/minimum-order-card/changeGroup.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @typedef { import(".~/data/actions").ShippingRate } ShippingRate
+ * @typedef { import("./typedefs").MinimumOrderGroup } MinimumOrderGroup
+ */
+
+/**
+ * Returns new shipping rates value, with updated group.
+ *
+ * Will set `shipingRate.options.free_shipping_threshold` to `newGroup.threshold`
+ * for the shipping rate countries that are listed in `newGroup.countries` (if provided),
+ * and set `threshold` to `undefined` for the countries which were in `oldGroup.countries` (if provided),
+ * but not in the new.
+ *
+ * @param {Array<ShippingRate>} value Shipping rates.
+ * @param {MinimumOrderGroup} [oldGroup] Old minimum order group.
+ * @param {MinimumOrderGroup} [newGroup] New minimum order group.
+ */
+export const changeGroup = ( value, oldGroup, newGroup ) => {
+	return value.map( ( shippingRate ) => {
+		const newShippingRate = {
+			...shippingRate,
+			options: {
+				...shippingRate.options,
+			},
+		};
+
+		if (
+			newGroup &&
+			newGroup.countries.includes( newShippingRate.country )
```

Comment:
💅 

`newGroup?.countries.includes( newShippingRate.country )`

and

`oldGroup?.countries.includes( newShippingRate.country )`

also works